eclipse-collections-code-generator/src/main/resources/api/map/objectPrimitiveMap.stg

```diff
@@ -94,6 +94,31 @@ public interface Object<name>Map\<K> extends <name>Iterable
     void forEachKeyValue(Object<name>Procedure\<? super K> procedure);
     <if(!primitive.booleanPrimitive)><(flipUniqueValues.(name))(name)><endif>
 
+    /**
+     * This method can be used to implement anySatisfy, allSatisfy, and noneSatisfy.
+     *
+     * anySatisfy  = this.shortCircuitKeyValue(predicate, true, true, false);
+     * allSatisfy  = this.shortCircuitKeyValue(predicate, false, false, true);
+     * noneSatisfy = this.shortCircuitKeyValue(predicate, true, false, true);
+     *
+     * @since 12.0
+     */
+    default boolean shortCircuitKeyValue(
```

Comment:
I think it's odd to have this method as `public` and as `default`.

`public` because it's an implementation detail for performance.

`default` because the implementation at this level creates wrapper objects and isn't there for performance.

UnifiedMap/UnifiedSet's shortCircuit() methods are private. But they take values, not key-value pairs.

I've had this idea for a MapView for a long time that I never filed as an issue. It might make sense to add at this point. I'll show what it would look like in AbstractHashBag if it existed in another comment.

Reply:
Agreed. I didn't like having this method on the API. I went back and forth with having any, all, none and whether to make this private on the interface, which works but causes a lot more generated code to review.

In the end, for the initial PR, I wanted to only show that there was a performance improvement with the minimal code necessary so that a change here was worthwhile.